server.go

```diff
@@ -12,38 +12,44 @@ import (
 	"google.golang.org/protobuf/reflect/protoreflect"
 )
 
-// requestHandler is used to fetch a response message based on the request.
-// A requestHandler should receive a message from the server, unmarshal it into
+// RequestHandler is used to fetch a response message based on the request.
+// A RequestHandler should receive a message from the server, unmarshal it into
 // the proper type for that Method's request type, call a user provided Handler,
 // and return a marshaled result to the server.
-type requestHandler func(ServerCtx, *Message, chan<- *Message)
+type (
+	Interceptor func(ServerCtx, *Message, Handler) (*Message, error)
+	Handler     func(ServerCtx, *Message) (*Message, error)
+)
 
-type orderingServer struct {
-	handlers map[string]requestHandler
-	opts     *serverOptions
-	ordering.UnimplementedGorumsServer
+type (
+	orderingServer struct {
+		handlers map[string]Handler
+		opts     *serverOptions
+		ordering.UnimplementedGorumsServer
+	}
+)
+
+func defaultInterceptor(ctx ServerCtx, msg *Message, next Handler) (*Message, error) {
+	// the default interceptor must be the outermost interceptor in the chain
+	// it releases the handler lock after the next handler returns
+	// allowing the next request to be processed
+	defer ctx.Release()
+	message, err := next(ctx, msg)
+	if message == nil && err == nil {
```

Comment:
Should we add a comment to explain what scenario we may experience `message==nil` and why we don't want to invoke `SendMessage` for this case?